### DISCUSSION TITLE
jinja2/django IPAddress has no parent object

### DISCUSSION THREAD

**Discussion by mostdaysarebaddays** [netbox-community/netbox#15625] [Help Wanted!] [answered]
Trying to get IPs from parent prefix - this works fine via API - doesn't look like it's being exposed in django

API:

list(nb.ipam.ip_addresses.filter(parent=test.prefix, description="test"))
[10.40.240.13/32]

{{ipam.IPAddress.objects.filter(parent=test.prefix, description="test")}}

Error:

<class 'django.core.exceptions.FieldError'>

Cannot resolve keyword 'parent' into field. Choices are: address, assigned_object, assigned_object_id, assigned_object_type, assigned_object_type_id, bookmarks, comments, created, custom_field_data, description, dns_name, fhrpgroup, id, interface, journal_entries, last_updated, local_address, nat_inside, nat_inside_id, nat_outside, remote_address, role, services, status, tagged_items, tags, tenant, tenant_id, vminterface, vrf, vrf_id

Python version: 3.8.12
NetBox version: 3.6.4

**Comment by candlerb** [comment]
There is no directly 'parent' database relationship between IPAddress and Prefix.

If you want to do it in Django ORM, you need to duplicate what the Netbox REST API filters are doing. For example, the parent filter on IPAddress is [here](https://github.com/netbox-community/netbox/blob/v3.7.5/netbox/ipam/filtersets.py#L653).

You can some find some more examples in [this script](https://github.com/netbox-community/customizations/pull/94/files) (not yet merged).

**Reply by mostdaysarebaddays** [reply]
Thanks, I'll look into it!

Would this be reasonable for a feature request?

**Reply by candlerb** [reply, marked as answer]
That's part of Django, I don't think you can hack the Django filters (easily).  In any case, what you want is already available, just under a different name:

```
ipam.IPAddress.objects.filter(address__net_host_contained=test.prefix, description="test")
```

**Reply by mostdaysarebaddays** [reply]
Perfect, thank you!
